rest-client-common/src/main/java/io/apicurio/rest/client/auth/OidcAuth.java

```diff
@@ -70,8 +79,15 @@ private void requestAccessToken() {
             ).collect(Collectors.joining("&"));
             final AccessTokenResponse accessTokenResponse = apicurioHttpClient.sendRequest(TokenRequestsProvider.obtainAccessToken(paramsEncoded));
             this.cachedAccessToken = accessTokenResponse.getToken();
-            this.cachedAccessTokenExp = System.currentTimeMillis() / 1000 + accessTokenResponse.getExpiresIn();
-
+            /**
+             * expiresIn is in seconds
+             */
+            Duration expiresIn = Duration.ofSeconds(accessTokenResponse.getExpiresIn());
```

Comment:
I would need to debug but I guess this is not correct since tests are not passing. Also, I would recommend using the methods `minusSeconds` and `plusSeconds` even if the behaviour is the same, it will allow you to use primitives for e.g. the default reduction time.

Reply:
default reduction time is enforced in the constructor

Reply:
That was my point. We're enforcing 20s in the tests and they're failing with the token not being expired.

Reply:
I can change it to this `if (expiresIn.compareTo(this.tokenExpirationReduction) >= 0) {` 
but then in the `AuthTest#oidcAuthTest` line 44 tokens won't be reused because the token will expire in 0 seconds...

Reply:
I have fixed the issue by setting a proper reduction in tests, should be good to go now.

Reply:
actually the change to `if (expiresIn.compareTo(this.tokenExpirationReduction) >= 0) {` makes sense